runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -123,214 +195,86 @@ public Dimension render(Graphics2D g)
+		BarRenderer right = barRenderers.get(config.rightBarMode());
 
-		final int maxHealth = client.getRealSkillLevel(Skill.HITPOINTS);
-		final int maxPrayer = client.getRealSkillLevel(Skill.PRAYER);
-		final int currentHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
-		final int currentPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
-		final int quickPrayerState = client.getVar(Varbits.QUICK_PRAYER);
-		final Color prayerBar = quickPrayerState == 1 ? QUICK_PRAYER_COLOR : PRAYER_COLOR;
+		if (left != null)
+			left.draw(g, offsetLeftBarX, offsetLeftBarY, height);
+		if (right != null)
+			right.draw(g, offsetRightBarX, offsetRightBarY, height);
 
-		renderBar(g, offsetHealthX, offsetHealthY,
-			maxHealth, currentHealth, height, healthBar);
+		return null;
+	}
 
-		renderBar(g, offsetPrayerX, offsetPrayerY,
-			maxPrayer, currentPrayer, height, prayerBar);
+	private int getRestoreValue(String skill)
+	{
+		final MenuEntry[] menu = client.getMenuEntries();
+		final int menuSize = menu.length;
+		final MenuEntry entry = menuSize > 0 ? menu[menuSize - 1] : null;
+		int restoreValue = 0;
 
-		if (config.enableRestorationBars())
+		if (entry != null && entry.getParam1() == WidgetInfo.INVENTORY.getId())
 		{
-			final MenuEntry[] menu = client.getMenuEntries();
-			final int menuSize = menu.length;
-			final MenuEntry entry = menuSize > 0 ? menu[menuSize - 1] : null;
-			int prayerHealValue = 0;
-			int foodHealValue = 0;
-			if (entry != null && entry.getParam1() == WidgetInfo.INVENTORY.getId())
+			final Effect change = itemStatService.getItemStatChanges(entry.getIdentifier());
+
+			if (change != null)
 			{
-				final Effect change = itemStatService.getItemStatChanges(entry.getIdentifier());
+				final StatsChanges statsChanges = change.calculate(client);
 
-				if (change != null)
+				for (final StatChange c : statsChanges.getStatChanges())
 				{
-					final StatsChanges statsChanges = change.calculate(client);
+					final String strVar = c.getTheoretical();
 
-					for (final StatChange c : statsChanges.getStatChanges())
+					if (Strings.isNullOrEmpty(strVar))
 					{
-						final String strVar = c.getTheoretical();
-
-						if (Strings.isNullOrEmpty(strVar))
-						{
-							continue;
-						}
-
-						final Integer value = Ints.tryParse(strVar.startsWith("+") ? strVar.substring(1) : strVar);
-
-						if (value == null)
-						{
-							continue;
-						}
+						continue;
+					}
 
-						if (c.getStat().getName().equals(Skill.HITPOINTS.getName()))
-						{
-							foodHealValue = value;
-						}
+					final Integer value = Ints.tryParse(strVar.startsWith("+") ? strVar.substring(1) : strVar);
 
-						if (c.getStat().getName().equals(Skill.PRAYER.getName()))
-						{
-							prayerHealValue = value;
-						}
+					if (value == null)
+					{
+						continue;
+					}
 
-						if (foodHealValue != 0 && prayerHealValue != 0)
-						{
-							break;
-						}
+					if (c.getStat().getName().equals(skill))
```

Comment:
Why isn't this a `continue` guard clause at the top of the loop? This can be the first thing checked before starting to parse `c.getTheoretical()`.